src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java

```diff
@@ -39,85 +40,148 @@ public ModelingSubmissionService(ModelingSubmissionRepository modelingSubmission
+                }
+            } catch (Exception e) {
+                throw new ConflictException("Conflict exception", "Tried to call createModelingSubmission() more than once for the same participation");
+            }
         } else {
+            // an unrated result was found for the participation
             result = optionalResult.get();
         }
 
-        result.setCompletionDate(ZonedDateTime.now());
-        modelingSubmission.setSubmissionDate(result.getCompletionDate());
-
-        if (modelingSubmission.getModel() != null && !modelingSubmission.getModel().isEmpty()) {
-            jsonModelRepository.writeModel(modelingExercise.getId(), participation.getStudent().getId(), modelingSubmission.getId(), modelingSubmission.getModel());
+        if (result.getId() == null) {
+            // there is no existing result and new result could not be created
+            return null;
         }
 
+        // update submission properties
+        modelingSubmission.setSubmissionDate(ZonedDateTime.now());
+        modelingSubmission.setType(SubmissionType.MANUAL);
+
         result.setSubmission(modelingSubmission);
         modelingSubmission.setResult(result);
         modelingSubmissionRepository.save(modelingSubmission);
         resultRepository.save(result);
 
+        if (modelingSubmission.getModel() != null && !modelingSubmission.getModel().isEmpty()) {
+            jsonModelRepository.writeModel(modelingExercise.getId(), user.getId(), modelingSubmission.getId(), modelingSubmission.getModel());
+        }
+
         if (modelingSubmission.isSubmitted()) {
             submit(modelingSubmission, modelingExercise);
+        } else if (modelingExercise.getDueDate() != null && !modelingExercise.isEnded()) {
+            // save submission to HashMap if exercise not ended yet
+            AutomaticSubmissionService.updateSubmission(modelingExercise.getId(), user.getLogin(), modelingSubmission);
         }
 
         return result;
     }
 
+    /**
+     * Adds a model to compass service to include it in the automatic grading process.
+     *
+     * @param modelingSubmission    the submission which contains the model
+     * @param modelingExercise      the exercise the model belongs to
+     */
     public void submit(ModelingSubmission modelingSubmission, ModelingExercise modelingExercise) {
         this.compassService.addModel(modelingExercise.getId(), modelingSubmission.getId(), modelingSubmission.getModel());
     }
 
+    /**
+     * Find the model for given exerciseId, studentId and model id, which is the same as submission id.
+     *
+     * @param exerciseId    the exercise id for which to find the model
+     * @param studentId     the student id for which to find the model
+     * @param id            the model id
+     * @return the model JsonObject if found otherwise null
+     */
     public JsonObject getModel(long exerciseId, long studentId, long id) {
```

Comment:
rename id to modelId, also describe in the comment what the modelId actually is?!?

Reply:
please also do according changes in the jsonModelRepository if necessary